packages/ui/components/overlays/src/OverlaysManager.js

```diff
@@ -108,7 +116,14 @@ export class OverlaysManager {
     if (!this.list.find(ctrl => ctrlToHide === ctrl)) {
       throw new Error('could not find controller to hide');
     }
+    // eslint-disable-next-line no-param-reassign
+    ctrlToHide._blocksEscKeyHandler = false;
     this.__shownList = this.shownList.filter(ctrl => ctrl !== ctrlToHide);
+    setTimeout(() => {
```

Comment:
Can we explain what this code does and why the timeout is needed?

Reply:
It removes the `_blocksEscKeyHandler` from the next overlay.
If there is no timeout, the unblocking will be done before the `Esc` will trigger the hide overlay. If you unblock it one tick later, it all works as expected.